deepinv/sampling/langevin.py

```diff
@@ -296,75 +68,37 @@ def __init__(
         g_statistic=lambda x: x,
         verbose=False,
     ):
-        iterator = ULAIterator(step_size=step_size, alpha=alpha, sigma=sigma)
+        params_algo = {"step_size": step_size, "alpha": alpha, "sigma": sigma}
+        iterator = ULAIterator(clip=clip)
         super().__init__(
             iterator,
-            prior,
             data_fidelity,
+            prior,
             max_iter=max_iter,
+            params_algo=params_algo,
             thresh_conv=thresh_conv,
-            g_statistic=g_statistic,
             burnin_ratio=burnin_ratio,
-            clip=clip,
             thinning=thinning,
-            save_chain=save_chain,
+            history_size=save_chain,
             verbose=verbose,
         )
+        self.g_statistics = [g_statistic]
```

Comment:
The original idea of `self.g_statistics` is to have more than one statistic that could be computed during then sampling process. Nevertheless, the way it's implemented here it will be only one, given that `g_statistic` is a `Callable` and not a `list[Callable]`, and that you're building the list in this line. I suggest having as input parameter a `list[Callable]` that will be directly stored in `self.g_statistics`